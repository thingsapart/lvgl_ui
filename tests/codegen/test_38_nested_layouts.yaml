# A vertical flex container whose children are horizontal flex containers (a common pattern).
- type: obj
  layout: LV_LAYOUT_FLEX
  flex_flow: LV_FLEX_FLOW_COLUMN
  size: [250, 150]
  pad_all: 5
  pad_gap: 5
  center: []
  children:
    - type: obj # First row
      layout: LV_LAYOUT_FLEX
      flex_flow: LV_FLEX_FLOW_ROW
      width: 100%
      flex_grow: 1
      children:
        - type: button
          flex_grow: 1
    - type: obj # Second row
      layout: LV_LAYOUT_FLEX
      flex_flow: LV_FLEX_FLOW_ROW
      width: 100%
      flex_grow: 1
      children:
        - type: slider
          flex_grow: 1
