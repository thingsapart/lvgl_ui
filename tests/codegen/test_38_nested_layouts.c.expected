/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: obj_0 (obj)
    lv_obj_t* obj_0 = lv_obj_create(parent);

    lv_obj_set_style_layout(obj_0, LV_LAYOUT_FLEX, 0);
    lv_obj_set_style_flex_flow(obj_0, LV_FLEX_FLOW_COLUMN, 0);
    lv_obj_set_style_size(obj_0, 250, 150, 0);
    lv_obj_set_style_pad_all(obj_0, 5, 0);
    lv_obj_set_style_pad_gap(obj_0, 5, 0);
    lv_obj_center(obj_0);
    // unnamed: obj_1 (obj)
    lv_obj_t* obj_1 = lv_obj_create(obj_0);

    lv_obj_set_style_layout(obj_1, LV_LAYOUT_FLEX, 0);
    lv_obj_set_style_flex_flow(obj_1, LV_FLEX_FLOW_ROW, 0);
    lv_obj_set_style_width(obj_1, lv_pct(100), 0);
    lv_obj_set_style_flex_grow(obj_1, 1, 0);
    // unnamed: button_2 (button)
    lv_obj_t* button_2 = lv_button_create(obj_1);

    lv_obj_set_style_flex_grow(button_2, 1, 0);


    // unnamed: obj_3 (obj)
    lv_obj_t* obj_3 = lv_obj_create(obj_0);

    lv_obj_set_style_layout(obj_3, LV_LAYOUT_FLEX, 0);
    lv_obj_set_style_flex_flow(obj_3, LV_FLEX_FLOW_ROW, 0);
    lv_obj_set_style_width(obj_3, lv_pct(100), 0);
    lv_obj_set_style_flex_grow(obj_3, 1, 0);
    // unnamed: slider_4 (slider)
    lv_obj_t* slider_4 = lv_slider_create(obj_3);

    lv_obj_set_style_flex_grow(slider_4, 1, 0);



}
